Ppx_expect doesn't seem to work on windows

I'm trying to make expect_test work on windows for js_of_ocaml.
Here is what I get in https://github.com/ocsigen/js_of_ocaml/pull/1132

using `ppx_expect.v0.14.1`, `ppx_inline_test.v0.14.1`

`cat debug-expect/dune`

```
(library
 (name debug_expect)
 (inline_tests
  (modes js native))
 (preprocess (pps ppx_expect)))
```

`cat debug-expect/debug_expect.ml`

```ocaml
let%expect_test _ =
  print_endline "hello";
  [%expect {| hello |}]
```

```
Run opam exec -- dune runtest debug-expect
  opam exec -- dune runtest debug-expect
  shell: C:\Program Files\PowerShell\7\pwsh.EXE -command ". '{0}'"
  env:
    OPAMCOLOR: always
    OPAMERRLOGLEN: 0
    OPAMJOBS: 2
    OPAMPRECISETRACKING: 1
    OPAMSOLVERTIMEOUT: 1000
    OPAMVERBOSE: false
    OPAMYES: 1
    OPAMROOT: D:\.opam
    HOME: C:\Users\runneradmin
    MSYS: winsymlinks:native
    CYGWIN: winsymlinks:native
    CYGWIN_ROOT: D:\cygwin
    CYGWIN_ROOT_BIN: D:\cygwin\bin
    CYGWIN_ROOT_WRAPPERBIN: D:\cygwin\wrapperbin
    DUNE_CACHE: enabled
    DUNE_CACHE_TRANSPORT: direct
inline_test_runner_debug_expect alias debug-expect/runtest (exit 2)
(cd _build/default/debug-expect && .debug_expect.inline-tests/inline_test_runner_debug_expect.exe inline-test-runner debug_expect -source-tree-root .. -diff-cmd -)
File "debug-expect/debug_expect.ml", line 1, characters 0-70 threw End_of_file.
  Raised at Base__Exn.protectx in file "src/exn.ml", line 71, characters 4-114
  Called from Ppx_inline_test_lib__Runtime.time_and_reset_random_seeds in file "runtime-lib/runtime.ml", line 356, characters 15-52
  Called from Ppx_inline_test_lib__Runtime.test in file "runtime-lib/runtime.ml", line 444, characters 52-83

FAILED 1 / 1 tests
Error: Process completed with exit code 1.
```

--- a/collector/expect_test_collector.ml
+++ b/collector/expect_test_collector.ml
@@ -101,7 +101,7 @@ module Make (C : Expect_test_config_types.S) = struct
 
     let create () =
       let filename = Filename.temp_file "expect-test" "output" in
-      let chan = open_out filename in
+      let chan = open_out_bin filename in
       before_test ~output:chan ~stdout ~stderr;
       { chan; filename = File.Name.of_string filename; saved = [] }
     ;;
@@ -120,7 +120,7 @@ module Make (C : Expect_test_config_types.S) = struct
     let relative_filename t = File.Name.relative_to ~dir:(File.initial_dir ()) t.filename
 
     let with_ic fname ~f =
-      let ic = open_in fname in
+      let ic = open_in_bin fname in
       protect ~finally:(fun () -> close_in ic) ~f:(fun () -> f ic)
     ;;
 

